### ISSUE TITLE
Support clipboard write() for audio formats (midi, wav)

### ISSUE THREAD

**Issue by chr15m** [w3c/clipboard-apis#218]
Hello! I am a developer of web based audio tools like https://dopeloop.ai/melody-generator and https://pro.sfxr.me. I am working on more tools at the moment for generating samples as well as MIDI.

Something users have asked me for is the ability to copy MIDI data directly from the page so that they can paste into their DAW such as Abelton, Garage Band, etc. At the moment I can't do this because the clipboard API allowed mimetypes don't include MIDI data.

I am currently working on a drum machine and it would likewise be great to let users copy drum loops and individual drum samples (as WAV formatted blobs) into their buffer to paste into their DAWs.

At the moment both of these use cases require a round trip through downloading the data to a local file (with .mid or .wav extension respectively) and then the user must import that file into their DAW or music app. The feedback I get is this is cumbersome. It would be a great usability enhancement to let them copy audio data directly from web apps so they can paste them into their DAWs, just like they can currently do with PNG files and image editors.

Thank you for your time!

P.S. Would love to hear @cwilso's opinion on this.

**Comment by tomayac** [comment]
I love the user case. This is more of a browser support request than a spec issue, though. For Chrome, the Async Clipboard API is being worked on in the context of Project Fugu 🐡. You can [request new Fugu features](https://bit.ly/new-fugu-request) through our bug tracker. For orientation, this is a [similar feature request](https://issues.chromium.org/issues/40808084) where support for a new image format is requested. If you [send me](mailto:tomac@google.com) the bug URL, I'm happy to help triage it. It's pretty much the same flow for the other browsers as well. 

(In Chromium-based browsers, there's [web custom formats](https://developer.chrome.com/blog/web-custom-formats-for-the-async-clipboard-api/) that allow you to copy anything, but it depends on support for the web custom format on the receiving end, that is, Abelton and friends.)

**Comment by saschanaz** [comment]
It would be nice to speak based on the specification rather than Chromium implementation here, for the interop sake.

Custom formats are not really suitable if the purpose is for interoperability with other (native) apps, right?

**Comment by saschanaz** [comment]
There are [mandatory/optional types](https://w3c.github.io/clipboard-apis/#mandatory-data-types-x) so I see at least some level of assumption. This issue can be read as a request to add a new item in the list.

**Comment by chr15m** [comment]
Yes it is a request to add audio mimetypes to the mandatory list. At least `audio/wav` and `audio/midi`.

I personally don't use Chrome as my daily browser, but obviously many users of my software do.

**Comment by tomayac** [comment]
> There are [mandatory/optional types](https://w3c.github.io/clipboard-apis/#mandatory-data-types-x) so I see at least some level of assumption. This issue can be read as a request to add a new item in the list.

That's fair, albeit there's [`ClipboardItem.supports()`](https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem/supports_static), so there's a baseline assumption that you can't know until you know—outside of the mandatory types. Since the [list has shrunk](https://github.com/w3c/clipboard-apis/pull/155) in the past, I _think_ there's limited interest in growing it again.

**Comment by saschanaz** [comment]
The new item indeed would only make sense when we get enough implementor interest.

**Comment by chr15m** [comment]
> get enough implementor interest

Cool. What does this mean in practice? What's the best way for me to make that happen? File issues on each browser as suggested previously in the issue?
